infra/main.tf

```diff
@@ -0,0 +1,508 @@
+}
+
+resource "google_sql_user" "db_root" {
+  name = "root"
+  instance = google_sql_database_instance.db.name
+  password = random_password.db_root_password.result
+}
+
+resource "kubernetes_secret" "database_server_config" {
+  metadata {
+    name = "database-server-config"
+  }
+
+  data = {
+    "server-ca.pem" = google_sql_database_instance.db.server_ca_cert.0.cert
+    "client-cert.pem" = google_sql_ssl_cert.root_client_cert.cert
+    "client-key.pem" = google_sql_ssl_cert.root_client_cert.private_key
+    "sql-config.cnf" = <<END
+[client]
+host=${google_sql_database_instance.db.ip_address[0].ip_address}
+user=root
+password=${random_password.db_root_password.result}
+ssl-ca=/sql-config/server-ca.pem
+ssl-cert=/sql-config/client-cert.pem
+ssl-key=/sql-config/client-key.pem
+ssl-mode=VERIFY_CA
+END
+    "sql-config.json" = <<END
+{
+    "docker_root_image": "gcr.io/${var.gcp_project}/ubuntu:18.04",
+    "host": "${google_sql_database_instance.db.ip_address[0].ip_address}",
+    "port": 3306,
+    "user": "root",
+    "password": "${random_password.db_root_password.result}",
+    "instance": "${google_sql_database_instance.db.name}",
+    "connection_name": "${google_sql_database_instance.db.connection_name}",
+    "ssl-ca": "/sql-config/server-ca.pem",
+    "ssl-cert": "/sql-config/client-cert.pem",
+    "ssl-key": "/sql-config/client-key.pem",
+    "ssl-mode": "VERIFY_CA"
+}
+END
+  }
+}
+
+resource "google_container_registry" "registry" {
+}
+
+resource "google_service_account" "gcr_pull" {
+  account_id   = "gcr-pull"
+  display_name = "pull from gcr.io"
+}
+
+resource "google_service_account_key" "gcr_pull_key" {
+  service_account_id = google_service_account.gcr_pull.name
+}
+
+resource "google_service_account" "gcr_push" {
+  account_id   = "gcr-push"
+  display_name = "push to gcr.io"
+}
+
+resource "google_service_account_key" "gcr_push_key" {
+  service_account_id = google_service_account.gcr_push.name
+}
+
+resource "google_storage_bucket_iam_member" "gcr_pull_viewer" {
+  bucket = google_container_registry.registry.id
+  role = "roles/storage.objectViewer"
+  member = "serviceAccount:${google_service_account.gcr_pull.email}"
+}
+
+resource "google_storage_bucket_iam_member" "gcr_push_admin" {
+  bucket = google_container_registry.registry.id
+  role = "roles/storage.admin"
+  member = "serviceAccount:${google_service_account.gcr_push.email}"
+}
+
+resource "kubernetes_secret" "gcr_pull_key" {
+  metadata {
+    name = "gcr-pull-key"
+  }
+
+  data = {
+    "gcr-pull.json" = base64decode(google_service_account_key.gcr_pull_key.private_key)
+  }
+}
+
+resource "kubernetes_secret" "gcr_push_key" {
+  metadata {
+    name = "gcr-push-service-account-key"
+  }
+
+  data = {
+    "gcr-push-service-account-key.json" = base64decode(google_service_account_key.gcr_push_key.private_key)
+  }
+}
+
+resource "kubernetes_namespace" "ukbb_rg" {
+  metadata {
+    name = "ukbb-rg"
+  }
+}
+
+resource "kubernetes_service" "ukbb_rb_browser" {
```

Comment:
I think this service might need a `depends_on = [kubernetes_namespace.ukbb_rg]` to avoid a race.

Reply:
Thanks!  I modified these entries to refer to the namespace resource to get the name, which should introduce the proper dependency.